srv/salt/ceph/tuned/mon/default.sls

```diff
@@ -0,0 +1,22 @@
+{% set roles = salt['pillar.get']('roles') %}
+{% set mgr_off = pillar.get('tuned_mgr_init', 'default') %}
+
+/etc/tuned/ses-mon/tuned.conf:
+  file.managed:
+    - source: salt://ceph/tuned/files/ses-mon/tuned.conf
+    - makedirs: True
+    - user: root
+    - group: root
+    - mode: 644
+
+{% if 'mgr' not in roles or 'default-off' not in mgr_off %}
```

Comment:
So you start tuned if mgr is not in roles or default-off is _not_ set. The former statement sounds weird to me. Is there a reason for this?

Reply:
I agree that it sounds weird, but it's true because the states `ceph.tuned.mgr`, `ceph.tuned.mon` and `ceph.tuned.osd` run sequentially and in this order.

If there's only one `tuned_<some role>_init` set to `default-off`, and `<some role>` is on this node, then we don't want to start `tuned`, even if there are other roles. Here's how it applies here: if `tuned_mgr_init` is set to `default-off` *and* there's a `mgr` role on the same node, it means that `ceph.tuned.mgr` (which ran first) turned `tuned` off, so we don't want to re-enable it in `ceph.tuned.mon`.

Only checking that `tuned_mgr_init` is not `default-off` to start `tuned` is not enough, because there might not be a `mgr` on this same node.

Again, this approach is based on the fact that the `tuned` states are applied sequentially in this order: `ceph.tuned.mgr`, `ceph.tuned.mon` and finally `ceph.tuned.osd`. This is also why this check isn't needed in `ceph/tuned/mgr/default.sls`. I hope this makes some sense. Also, the same thing applies for `ceph/tuned/osd/default.sls`.

Reply:
Thanks, got it!